app/controllers/beaches_controller.rb

```diff
@@ -1,6 +1,10 @@
 class BeachesController < ApplicationController
   before_action :authorize_user, except: [:index, :show]
 
+  def index
+    search(params[:query])
+  end
```

Comment:
Rails has some design patterns that are best practice to follow, like seeing `@modelname` in the index action rather than calling another method in it. You need to do two things here. Update the `index` method here . . . 

```ruby
def index
  if params[:query]
    @beaches = Beach.search(params[:query])
  else
    @beaches = Beach.all
end
```

. . . and change your `search` method into a class method in your `Beach` model so that you can call `Beach.search(params[:query])` and have it handle the DB fetching.